selfservice/strategy/oidc/strategy_registration.go

```diff
@@ -185,6 +185,32 @@ func (s *Strategy) Register(w http.ResponseWriter, r *http.Request, f *registrat
 	return errors.WithStack(flow.ErrCompletedByStrategy)
 }
 
+func (s *Strategy) registrationToLogin(w http.ResponseWriter, r *http.Request, rf *registration.Flow, providerID string) (*login.Flow, error) {
+	// If return_to was set before, we need to preserve it.
+	var opts []login.FlowOption
+	if len(rf.ReturnTo) > 0 {
+		opts = append(opts, login.WithFlowReturnTo(rf.ReturnTo))
+	}
+
+	if len(rf.UI.Messages) > 0 {
+		opts = append(opts, login.WithFormErrorMessage(rf.UI.Messages))
+	}
+
+	// This endpoint only handles browser flow at the moment.
+	lf, _, err := s.d.LoginHandler().NewLoginFlow(w, r, flow.TypeBrowser, opts...)
+
+	if err != nil {
+		return nil, s.handleError(w, r, rf, providerID, nil, err)
```

Comment:
All parameters seem to be supplied by the parameters to this function. So you should be able to remove this `s.handleError` call in here, and add it to the part that **calls** this function.

Reply:
Make sense, have updated it :+1: